### PR TITLE
[9.0][ADD] medical_base_history: Module to track historical changes

### PR DESCRIPTION
**Pull request by lasley**
* Create module to track changes on specific records over time

Odoo Medical History
====================

Extension of medical that provides an abstract history object for medical record reporting requirement compliance

cc @tedsalmon @jcdrubay

### CODE REVIEW

File: medical_base_history/README.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+* Improve and provide a full description for this module into the README.rst
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/vertical-medical/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
```
Comment (tedsalmon) [inline medical_base_history/README.rst:34]: Using the old README template ;P

File: medical_base_history/models/medical_history_abstract.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    to avoid name collisions with models that will inherit from this class.
+
+    Attributes:
+        _audit_on: `list` of methods to log. Supports create, write, and delete
+            Override this in child classes in order to add/remove attrs
+    """
+
+    _name = 'medical.history.abstract'
+    _description = 'Provides inheritable model for change history auditing'
+    _audit_on = ['create', 'write', 'delete', ]  # @TODO: Read
```
Comment (tedsalmon) [inline medical_base_history/models/medical_history_abstract.py:21]: Move to Roadmap

File: medical_base_history/models/medical_history_type.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api
+from openerp.exceptions import ValidationError
+
+
+class MedicalHistoryType(models.Model):
+    '''
```
Comment (tedsalmon) [inline medical_base_history/models/medical_history_type.py:10]: PEP-0257 - s/'''/"""

File: medical_base_history/tests/test_medical_history_entry.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    #         expect, got,
+    #     )
+    #
+    # @mock.patch('%s.pickle' % entry_mdl)
+    # def test_write_old_record_dict_calls_pickle_dumps_with_rec(self, mk):
+    #     self.entry_type_id.new_cols_to_save = 'all'
+    #     self._new_entry()
+    #     mk.dumps.assert_called_with(self.vals)
+
+    # @TODO: Figure out how to test attr assignment without triggering compute
```
Comment (tedsalmon) [inline medical_base_history/tests/test_medical_history_entry.py:73]: Move to roadmap

File: medical_base_history/models/medical_history_type.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                'suffix': self.suffix,
+            }
+
+    @api.multi
+    @api.constrains('code')
+    def _check_unique_code(self):
+        ''' Constrain that code is unique '''
+        for rec_id in self:
+            if len(self.search([('code', '=', rec_id.code)])) > 1:
+                raise ValidationError(
```
Comment (tedsalmon) [inline medical_base_history/models/medical_history_type.py:70]: i10n required, per lint

**Review by tedsalmon — CHANGES_REQUESTED** [review]
Looks good minus a few requested changes.

**Comment by lasley** [comment]
We have deprecated this module for v10, and I think it's actually probably best to do the same for 9. We will backport LasLabs#171 to #70

**Pull request closed without merging**